### ISSUE TITLE
WC PB API returns actual block reference (extremely unsafe)

### ISSUE THREAD

**Issue by Blaylock1988** [sstixrud/WeaponCore#220]
## Is this a multiplayer or singleplayer bug?
Both

## Describe the bug:
WC PB API returns actual block reference, which allows any person's script to manipulate a targeted grid's blocks. This can be easily abused by griefers and hackers.

## Steps To Reproduce:
Can provide a script from SgtMook if desired to use as example. Please let me know if desired.

## Expected behavior:
Would be safer if it returned MyDetectedEntityInfo, as this is what vanilla turrets and camera raycasts return, and most scripts expect this method anyways. This would prevent unwanted script abuse.

### User Error Report (if Weapon-Specific) and log files
*WeaponCore lets you generate an error report for a specific weapon, while ingame, hold Ctrl+R and aim the red line at the Weapon that is erroring, and an error report will be generated in WeaponCores log cache: `Users\UserName\AppData\Roaming\SpaceEngineers\Storage\1918681825.sbm_WeaponCore`

Zip up that folder and link it here:

**Comment by Starwaster** [comment]
Could another player's cockpit be disabled this way? Recently I was involved in a battle where I could activate systems via the toolbar but could not move (rotation nor thrusters)

The cockpit block was inexplicably grayed out as though 'show block in terminal' were off. There was also an option to toggle block, which I have never seen for the cockpit before; it's always toggled on. After toggling it, I was able to move as normal.

The option to toggle block is gone and I have not seen it since. I chalked it up to just a normal random bug except that all my other faction-mates had similar issues... almost all combat vehicles were affected.
